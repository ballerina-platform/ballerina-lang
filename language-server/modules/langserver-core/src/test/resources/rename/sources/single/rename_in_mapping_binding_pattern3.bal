type Person record {|
    string name;
    int id;
|};

function test1() {
    Person p = {
        name: "John Doe",
        id: 1
    };
    
    var {name, id} = p;
    int size = name.length();
}
