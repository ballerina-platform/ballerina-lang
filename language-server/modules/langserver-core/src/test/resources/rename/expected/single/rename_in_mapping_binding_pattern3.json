{
  "source": {
    "file": "rename_in_mapping_binding_pattern3.bal"
  },
  "position": {
    "line": 12,
    "character": 18
  },
  "prepareRename": {
    "valid": true
  },
  "result": {
    "changes": {
      "rename_in_mapping_binding_pattern3.bal": [
        {
          "range": {
            "start": {
              "line": 11,
              "character": 9
            },
            "end": {
              "line": 11,
              "character": 13
            }
          },
          "newText": "name: pName"
        },
        {
          "range": {
            "start": {
              "line": 12,
              "character": 15
            },
            "end": {
              "line": 12,
              "character": 19
            }
          },
          "newText": "pName"
        }
      ]
    }
  }
}
